src/test/java/com/google/api/codegen/testdata/java_test_mock_service_impl_library.baseline

```diff
@@ -46,154 +47,229 @@ import java.util.List;
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void deleteBook(DeleteBookRequest request,
-       StreamObserver<Empty> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Empty.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void updateBook(UpdateBookRequest request,
-       StreamObserver<Book> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Book.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void moveBook(MoveBookRequest request,
-       StreamObserver<Book> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Book.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void listStrings(ListStringsRequest request,
-       StreamObserver<ListStringsResponse> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(ListStringsResponse.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void addComments(AddCommentsRequest request,
-       StreamObserver<Empty> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Empty.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void getBookFromArchive(GetBookFromArchiveRequest request,
-       StreamObserver<Book> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Book.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void updateBookIndex(UpdateBookIndexRequest request,
-       StreamObserver<Empty> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(Empty.newBuilder().build());
-     responseObserver.onCompleted();
-   }
-
-   @Override
-   public void streamShelves(ListShelvesRequest request,
-       StreamObserver<ListShelvesResponse> responseObserver) {
-     requests.add(request);
-     responseObserver.onNext(ListShelvesResponse.newBuilder().build());
-     responseObserver.onCompleted();
-   }
+  private ArrayList<GeneratedMessage> requests;
+  private ArrayList<GeneratedMessage> responses;
+
+  public MockLibraryServiceImpl() {
+    requests = new ArrayList<>();
+    responses = new ArrayList<>();
+  }
+
+  public List<GeneratedMessage> getRequests() {
+    return requests;
+  }
+
+  public void setResponses(List<GeneratedMessage> responses) {
+    this.responses = Lists.newArrayList(responses);
+  }
+
+  public void reset() {
+    requests = new ArrayList<>();
+    responses = new ArrayList<>();
+  }
+
+  @Override
+  public void createShelf(CreateShelfRequest request,
+      StreamObserver<Shelf> responseObserver) {
+    Shelf response = Shelf.newBuilder().build();
+    if (responses != null && !responses.isEmpty()) {
+      response = (Shelf) responses.get(0);
+    }
```

Comment:
Done.
After second thought I think you are right it is better to always require injected response since default response does not provide much testing value.